### DISCUSSION TITLE
How to plot and farm on seperate GPUs in same system?

### DISCUSSION THREAD

**Discussion by XxViper06xX** [madMAx43v3r/chia-gigahorse#229] [Q&A]
I have two GPUs available (A4000 and 3060Ti) and I would like to Plot with one and farm with the other in the same system. I think I've started to run into the limits of farming and plotting on the same GPU with C19 as the plotter randomly seems to crash.

I would like to plot with the A4000 and farm with a 3060Ti as for now one 3060Ti can handle my entire farm, per the chart provided by Max.

I feel like this is probably an easy thing to do, essentially bind the plotting to one and bind the farming to the other.

I can see how to assign or limit GPU in plotting so I can make the GPU I want be the plotter. But I do not see how to assign/limit the GPU used for farming.

Could running a VM be a simple solution? Maybe not the most eligant but possible?

My system is a T7610 with dual 2670 CPUs. 512GB Ram. Windows 10

Any help would be greatly appreciated.

**Comment by happycouak** [comment]
I think these topic are pretty well documented, here is a good starting point : https://github.com/madMAx43v3r/chia-gigahorse/blob/master/chiapos/README.md#limit-gpu-usage

Check the plotter doc also to limit to one GPU.

**Reply by XxViper06xX** [reply]
Ok so that README is farmer specific so then limiting the GPU used for the plotter I would use the -g argument. Correct?

**Reply by XxViper06xX** [reply]
So after tinkering with it for a bit I can't get CHIAPOS_MAX_CUDA_DEVICES=1 to use the 3060 for FARMING. Does this variable always choose the faster GPU or use the default or 0 device? My A4000 is GPU 0 and the 3060 is GPU 1. I can get Farming on the 3060 only by using CUDA_VISIBLE_DEVICES=1 but then in Plot Manager my A4000 is no longer an option for plotting.

**Reply by happycouak** [reply]
Have you tried CUDA_VISIBLE_DEVICES ?
